pkg/cluster/fixssh.go

```diff
@@ -100,14 +101,50 @@ func (m *manager) checkandUpdateNIC(ctx context.Context, resourceGroup string, i
 			}
 		}
 
-		if m.updateNIC(ctx, &nic, nicName, &lb, i, infraID) {
+		if m.updateILBAddressPool(ctx, &nic, nicName, &lb, i, infraID) {
+			m.log.Printf("updating Network Interface %s", nicName)
+			err = m.interfaces.CreateOrUpdateAndWait(ctx, resourceGroup, nicName, nic)
+			if err != nil {
+				return err
+			}
+		}
+
+		var elbName string
+		switch m.doc.OpenShiftCluster.Properties.ArchitectureVersion {
+		case api.ArchitectureVersionV1:
+			elbName = infraID + "-public-lb"
+		case api.ArchitectureVersionV2:
+			elbName = infraID
+		}
+
+		if m.doc.OpenShiftCluster.Properties.ArchitectureVersion == api.ArchitectureVersionV1 {
```

Comment:
Could this if statement be moved inside the archv1 switch statement?

Reply:
As discussed with you the **if** the statement moved inside the switch statement it makes the code less readable and adds little confusion even though the functionality does not does not change.

Reply:
an alternative would be to encapsulate the logic into a helper function, which it looks like you partially did. if the retrieval of the load balancers on line 121 were moved into the `updateV1ELBAddressPool` function then it would be fine to move this call into the switch. 

Also worth noting, is if `updateV1ELBAddressPool` returns an error instead of a `bool` the code that is actually doing the update can be moved into it making the switch on line 115 more in line with what @ellis-johnson suggested